checker/jdk/nullness/src/javax/script/ScriptEngineManager.java

```diff
@@ -0,0 +1,421 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package javax.script;
+import java.util.*;
+import java.security.*;
+import java.util.ServiceLoader;
+import java.util.ServiceConfigurationError;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * The <code>ScriptEngineManager</code> implements a discovery and instantiation
+ * mechanism for <code>ScriptEngine</code> classes and also maintains a
+ * collection of key/value pairs storing state shared by all engines created
+ * by the Manager. This class uses the <a href="../../../technotes/guides/jar/jar.html#Service%20Provider">service provider</a> mechanism to enumerate all the
+ * implementations of <code>ScriptEngineFactory</code>. <br><br>
+ * The <code>ScriptEngineManager</code> provides a method to return a list of all these factories
+ * as well as utility methods which look up factories on the basis of language name, file extension
+ * and mime type.
+ * <p>
+ * The <code>Bindings</code> of key/value pairs, referred to as the "Global Scope"  maintained
+ * by the manager is available to all instances of <code>ScriptEngine</code> created
+ * by the <code>ScriptEngineManager</code>.  The values in the <code>Bindings</code> are
+ * generally exposed in all scripts.
+ *
+ * @author Mike Grogan
+ * @author A. Sundararajan
+ * @since 1.6
+ */
+public class ScriptEngineManager  {
+    private static final boolean DEBUG = false;
+    /**
+     * The effect of calling this constructor is the same as calling
+     * <code>ScriptEngineManager(Thread.currentThread().getContextClassLoader())</code>.
+     *
+     * @see java.lang.Thread#getContextClassLoader
+     */
+    @SuppressWarnings("initialization")    // Until execution of this constructor, no such dereferencing problems will be there as illustrated in the manual 
```

Comment:
I don't understand this.  What manual?  What "such deferencing problems"?  Why is the comment talking about time before executing the constructior?